driver-core/src/test/java/com/datastax/driver/core/UnsetValuesTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.fail;
+
+import static com.datastax.driver.core.ProtocolVersion.V3;
+import static com.datastax.driver.core.ProtocolVersion.V4;
+
+public class UnsetValuesTest {
+
+    private CCMBridge ccm;
+    private Cluster cluster;
+    private PreparedStatement prepared;
+    private Session session;
+
+    @BeforeClass
+    public void createCcm() throws Exception {
+        ccm = CCMBridge.create("test", 1);
+    }
+
+    @AfterClass
+    public void closeCcm() throws Exception {
+        if(ccm != null) ccm.remove();
+    }
+
+    public void buildCluster(ProtocolVersion version) {
+        cluster = Cluster.builder()
+            .addContactPoint(CCMBridge.IP_PREFIX + 1)
+            .withProtocolVersion(version).build();
+        session = cluster.newSession();
+        session.execute("CREATE KEYSPACE IF NOT EXISTS ks WITH replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }");
+        session.execute("USE ks");
+        session.execute("CREATE TABLE IF NOT EXISTS foo (c1 int primary key, c2 text)");
+        prepared = session.prepare("INSERT INTO foo (c1, c2) VALUES (?, ?)");
+        BoundStatement st1 = prepared.bind();
+        st1.setInt(0, 42);
+        st1.setString(1, "foo");
+        session.execute(st1);
+    }
+
+    @AfterMethod
+    public void closeCluster() throws Exception {
+        if(cluster != null) cluster.close();
+    }
+
+    @Test(groups = "short")
+    public void should_not_allow_unset_value_on_bound_statement_when_protocol_lesser_than_v4() {
+        buildCluster(V3);
```

Comment:
This will fail when tested against C* 1.2 and 2.0. If you follow my recommendation of extending `PerClassSingleNodeCluster`, the default `cluster` has the highest possible version:

    cluster.getConfiguration().getProtocolOptions().getProtocolVersionEnum();

If that's < v4, you can use it for your test. Otherwise, build another instance with V3 forced and use that.